presto-main/src/main/java/com/facebook/presto/sql/planner/SortExpressionExtractor.java

```diff
@@ -55,8 +63,52 @@ private SortExpressionExtractor() {}
             return Optional.empty();
         }
 
-        if (filter instanceof ComparisonExpression) {
-            ComparisonExpression comparison = (ComparisonExpression) filter;
+        return new SortExpressionVisitor(buildSymbols).process(filter);
+    }
+
+    private static class SortExpressionVisitor
+            extends AstVisitor<Optional<SortExpressionContext>, Void>
+    {
+        private final Set<Symbol> buildSymbols;
+
+        public SortExpressionVisitor(Set<Symbol> buildSymbols)
+        {
+            this.buildSymbols = buildSymbols;
+        }
+
+        @Override
+        protected Optional<SortExpressionContext> visitExpression(Expression expression, Void context)
+        {
+            return Optional.empty();
+        }
+
+        @Override
+        protected Optional<SortExpressionContext> visitLogicalBinaryExpression(LogicalBinaryExpression binaryExpression, Void context)
+        {
+            if (binaryExpression.getType() != AND) {
+                return Optional.empty();
+            }
+            Optional<SortExpressionContext> leftProcessed = process(binaryExpression.getLeft());
+            Optional<SortExpressionContext> rightProcessed = process(binaryExpression.getRight());
+
+            if (!leftProcessed.isPresent() || !rightProcessed.isPresent() || !leftProcessed.get().getSortExpression().equals(rightProcessed.get().getSortExpression())) {
+                return Optional.empty();
```

Comment:
Return left (ie any) and put TODO to make it cost-based decision.
If there are two `<` conditions on two different build symbols, better to have sorted position links on one of them than on none.

Reply:
?